service/src/main_impl.rs

```diff
@@ -674,34 +681,46 @@ fn start_worker<E, T, D, InitializationHandler, WorkerModeProvider>(
 fn init_provided_shard_vault<E: EnclaveBase>(
 	shard: &ShardIdentifier,
 	enclave: &Arc<E>,
-	node_api: &ParentchainApi,
+	integritee_rpc_api: ParentchainApi,
+	maybe_target_a_rpc_api: Option<ParentchainApi>,
+	maybe_target_b_rpc_api: Option<ParentchainApi>,
+	shielding_target: Option<ParentchainId>,
 	we_are_primary_validateer: bool,
 ) {
+	let shielding_target = shielding_target.unwrap_or(ParentchainId::Integritee);
+	let rpc_api = match shielding_target {
+		ParentchainId::Integritee => integritee_rpc_api,
+		ParentchainId::TargetA => maybe_target_a_rpc_api
+			.expect("target A must be initialized to be used as shielding target"),
+		ParentchainId::TargetB => maybe_target_b_rpc_api
+			.expect("target B must be initialized to be used as shielding target"),
+	};
```

Comment:
Can't we just have an `Option<ShieldingTargetConfig>>`? I reckon that this will make things easier.

Reply:
I suggest to revisit the current design patterns around parentchain branching. I'm not claiming elegance in the current design 8-P